repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -72,27 +74,42 @@ public AkkaServiceClientImpl(HttpClientService httpClientService) {
                 .expireAfterWrite(FUTURE_CACHE_TTL, FUTURE_CACHE_UNIT)
                 .build();
 
-
-        tokenActorRef = actorSystem.actorOf(new Props(new UntypedActorFactory() {
-            public UntypedActor create() {
-
-                return new AuthTokenFutureActor(serviceClient);
-            }
-        }).withRouter(new RoundRobinRouter(numberOfActors)), "authRequestRouter");
+        createTokenActor();
     }
 
-
     @PreDestroy
     public void destroy() {
-        //Tie this actor system to our bean lifecycle
+        // tie this actor system to our bean lifecycle
         actorSystem.shutdown();
     }
 
     @Override
+    public void setConnectionPoolId(String connectionPoolId) {
+        if (StringUtils.equals(this.connectionPoolId, connectionPoolId)) {
+            return;
+        }
+
+        ActorRef oldTokenActorRef = tokenActorRef;
+        this.connectionPoolId = connectionPoolId;
+        createTokenActor();
+
+        if (oldTokenActorRef != null) {
+            actorSystem.stop(oldTokenActorRef);
+        }
+    }
+
+    private void createTokenActor() {
+        ServiceClient serviceClient = new ServiceClient(connectionPoolId, httpClientService);
+        this.timeout = new Timeout(serviceClient.getSocketTimeout() + CONNECTION_TIMEOUT_BUFFER_MILLIS, TimeUnit.MILLISECONDS);
+        this.tokenActorRef = actorSystem.actorOf(Props.create(AuthTokenFutureActor.class, serviceClient)
```

Comment:
I think we'll have to level up our abstraction here. We're probably going to have to have a parent actor that relays messages to the children so that we can recreate the children with a new connection PoolID. 

Or, create another actor that you can use to have all these children talk to to get that ServiceClient. Then you can poisonpill the router, and create a new one, after updating the connection pool provider router. 

There isn't a really clean way to do this, without adding more actors, and it's a bit difficult to describe without having a whiteboard :(